CONTRIBUTING.md

```diff
@@ -1,76 +1,268 @@
+git clone https://github.com/electron/electron.atom.io
+cd electron.atom.io
+npm install
+npm run dev
 ```
 
+You should now have an Express server running at 
+[localhost:5000](http://localhost:5000) :tada:
+
+Read on for more info about the structure of the site.
+
+- [Content](#content)
+  - [Documentation](#documentation)
+  - [Apps](#apps)
+  - [Blog](#blog)
+  - [Translations](#translations)
+- [Routes](#routes)
+- [Middlware](#middleware)
+- [Views](#views)
+- [Context](#context)
+- [Styles](#styles)
+- [Fontend JavaScript](#frontend-javascript)
+- [Scripts](#scripts)
+- [Environment Variables](#environment-variables)
+
+## Content
+
+The content on [electronjs.org] comes from a number of sources.
+
 ### Documentation
 
-The documentation on this site is pulled directly from the `electron/electron` repository's `docs` directory. Contributions to the documentation should be made there: [electron/electron](https://github.com/electron/electron/tree/master/docs).
+Electron's English API docs and tutorials live in the [electron/electron] repo 
+in the `docs` directory. Contributions to the documentation should be made there: [electron/electron].
+
+After being [translated](#translations) on Crowdin, all the docs are automatically
+updated on the [electron/electron-i18n] repo and auto-published to npm as the
+`electron-i18n` package. That package is in turne used by this website repo
+as the source of all documentation. See [electron/electron-i18n#usage](https://github.com/electron/electron-i18n#usage) for an explanation of the
+documentation data's structure.
+
+### Apps
+
+Want to add your app to the site? There's a repo for that. 
+See [electron/electron-apps](https://github.com/electron/electron-apps#adding-your-app).
+
+### Blog
+
+Blog posts live in this repo, in `/data/blog`.
 
-### Blog Posts
+Blogs are written in "Jekyll style", with YML frontmatter at the top for
+metadata. The `title`, `author`, and `date` properties should be set
+on all posts:
+
+```
+---
+title: 'Project of the Week: WebTorrent'
+author: zeke
+date: '2017-03-14'
+---
+```
 
 A few guidelines to keep in mind when publishing a blog post:
 
 * Posts should normally be published on Tuesday mornings (Pacific time) for maximum social reach.
 * Make sure the date in the filename is today's date
 * Tweet about it once the post is live!
 
-## How the Site is Built
+### Translations
+
+Electron's API documentation, tutorials, and website content are translated
+into other Languages on a third-party site called Crowdin. Crowdin is a 
+free and open tool, and  anyone is welcome to join in the translation process. 
+You can create an account easily and quickly by logging in with your GitHub 
+account.
+
+See 
+[electron/electron-i18n](https://github.com/electron/electron-i18n) for info 
+on how to get started, or jump right into translating at 
+[crowdin.com/project/electron](https://crowdin.com/project/electron).
+
+## Routes
+
+Routes are defined in [server.js](server.js).
```

Comment:
I know this might be oversimplifying, but should we specify what we mean by routes, eg:
Website routes are defined in [server.js](server.js).